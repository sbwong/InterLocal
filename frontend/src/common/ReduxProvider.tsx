import React, { ReactNode } from "react";

import { Provider } from "react-redux";
import { store } from "../app/store";

export interface ReduxProviderProps {
	children: ReactNode;
}

export default function ReduxProvider(props: ReduxProviderProps) {
	return <Provider store={store}>{props.children}</Provider>;
}
