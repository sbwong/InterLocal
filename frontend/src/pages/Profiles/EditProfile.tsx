import Box from "@material-ui/core/Box";
import EditPasswordHeader from "./components/EditPasswordHeader";
import EditProfileHeader from './components/EditProfileHeader';
import React from 'react';

export function EditProfile() {

    return (
        <Box paddingTop="30px" marginTop="50px" marginBottom="100px" justifyContent="center">
            <div>
                <EditProfileHeader />
                <EditPasswordHeader />
            </div>
        </Box>
  );
}
