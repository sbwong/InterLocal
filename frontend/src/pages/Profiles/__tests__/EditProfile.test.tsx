import React from "react";
import ReduxProvider from "../../../common/ReduxProvider";
import ReactDOM from "react-dom";
import { EditProfile } from "../EditProfile";
import { render } from "@testing-library/react";

it("renders without crashing", () => {
	const div = document.createElement("div");

	ReactDOM.render(
		<ReduxProvider>
			<EditProfile />
		</ReduxProvider>,
		div
	);
});

it("renders EditProfile page", () => {
	const { editprofile } = render(
		<ReduxProvider>
			<EditProfile />
		</ReduxProvider>
	);
	expect(editprofile).toMatchSnapshot();
});
