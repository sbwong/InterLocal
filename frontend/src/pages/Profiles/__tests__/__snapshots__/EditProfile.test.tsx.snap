// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders EditProfile page 1`] = `undefined`;
